### PR TITLE
Alternative hook and settings

### PR DESCRIPTION
**Pull request by GDmac**
- Setting for alternative (custom) hook
- Setting to limit queueing to certain channels
- Allow to reset queue from CP (e.g. after changing settings)

### CODE REVIEW

**Comment by GDmac** [comment]
When i first installed scheduled trigger, it queued all entries from an agenda/event channel as well. And as the queue triggers entry_submission_end with "new" entries, it would have send mail notifications again, all over the place.

(ps. i use scheduled_trigger as the perfect addition to auto_expire, which just updates the database)

**Comment by leigeber** [comment]
Awesome pull request. The latter 2 items were on our to-do list so this is great. I'll review everything later today, pull and then follow up with a push to get everything massaged into our internal PHP style guide. I hope this addon can be useful for a lot of people needing similar functionality.

**Comment by GDmac** [comment]
For my purposes (MX_notify) the channel selection settings was actually enough, but i can imagine other extensions hanging on to submission_end and maybe causing troubles/havoc.

Yes, this needs to be refactored *cough*.  Some attention points:
- check if calling multiple triggers causes no problems (double add to queue etc.)
- e.g. maybe change trigger checkboxs to input radio or single select for the trigger setting,
- maybe even per channel trigger setting.
- some code is not entirely DRY (duplicates), like channel checks and setting $now

** **edit** **
- the check_date in model is fetched several times, constructor should be enough
- maybe insert check_date instead of update, this allows monitoring the cron process

File: system/expressionengine/third_party/scheduled_trigger/views/scheduled_trigger_queue.php [hunk 1]
```diff
@@ -16,7 +16,7 @@
 		<?php foreach ($queue as $row) { ?>
 		<tr>
 			<td><?=$row['entry_id'] ?></td>
-			<td><a href="<?=BASE ?>&D=cp&C=content_publish&M=entry_form&entry_id=<?=$row['entry_id'] ?>"><?=$row['title'] ?></a></td>
+			<td><a href="<?=BASE;?>&amp;C=content_publish&amp;M=entry_form&amp;entry_id=<?=$row['entry_id'];?>"><?=$row['title'] ?></a></td>
```
Comment (GDmac) [inline system/expressionengine/third_party/scheduled_trigger/views/scheduled_trigger_queue.php:19]: ps. this fixes links to entries from the queue view. (tested in EE 2.8, maybe needs to be tested on older versions)

**Pull request closed without merging**